### ISSUE TITLE
PHP 8.1 compatibility

### ISSUE THREAD

**Issue by mlocati** [mongodb/mongo-php-library#850]
It's not possible to compile the mongodb php extension for PHP 8.1.

Here's the relevant output of the compilation log:

```
cc [...]  -c src/MongoDB/BulkWrite.c -MMD -MF src/MongoDB/BulkWrite.dep -MT src/MongoDB/BulkWrite.lo  -fPIC -DPIC -o src/MongoDB/.libs/BulkWrite.o
In file included from mongodb-1.10.0/src/MongoDB/BulkWrite.c:26:
mongodb-1.10.0/src/MongoDB/BulkWrite.c: In function 'php_phongo_bulkwrite_init_ce':
mongodb-1.10.0/php_phongo.h:174:21: error: 'zend_class_serialize_deny' undeclared (first use in this function); did you mean 'zend_ce_serializable'?
   ce->serialize   = zend_class_serialize_deny;   \
                     ^~~~~~~~~~~~~~~~~~~~~~~~~
mongodb-1.10.0/src/MongoDB/BulkWrite.c:669:2: note: in expansion of macro 'PHONGO_CE_DISABLE_SERIALIZATION'
  PHONGO_CE_DISABLE_SERIALIZATION(php_phongo_bulkwrite_ce);
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
mongodb-1.10.0/php_phongo.h:174:21: note: each undeclared identifier is reported only once for each function it appears in
   ce->serialize   = zend_class_serialize_deny;   \
                     ^~~~~~~~~~~~~~~~~~~~~~~~~
mongodb-1.10.0/src/MongoDB/BulkWrite.c:669:2: note: in expansion of macro 'PHONGO_CE_DISABLE_SERIALIZATION'
  PHONGO_CE_DISABLE_SERIALIZATION(php_phongo_bulkwrite_ce);
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
mongodb-1.10.0/php_phongo.h:175:21: error: 'zend_class_unserialize_deny' undeclared (first use in this function); did you mean 'zend_unserialize_data'?
   ce->unserialize = zend_class_unserialize_deny; \
                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~
mongodb-1.10.0/src/MongoDB/BulkWrite.c:669:2: note: in expansion of macro 'PHONGO_CE_DISABLE_SERIALIZATION'
  PHONGO_CE_DISABLE_SERIALIZATION(php_phongo_bulkwrite_ce);
```

I'm not sure how to fix this. Maybe https://github.com/igbinary/igbinary/blob/3.2.4/src/php7/igbinary.c#L112-L122 could be the solution?

**Comment by jmikola** [comment]
FYI: You opened this issue on the library repository, but the extension has its own repository and issue tracker in [mongodb/mongo-php-driver](https://github.com/mongodb/mongo-php-driver). The [new issue template](https://github.com/mongodb/mongo-php-library/issues/new/choose) also calls attention to that.

In any event, PHP 8.1 compatibility is currently being tracked in [PHPC-1850](https://jira.mongodb.org/browse/PHPC-1850) and there is a ticket specifically for the serialization issue ([PHPC-1849](https://jira.mongodb.org/browse/PHPC-1849)). Thanks for pointing out the fix in https://github.com/igbinary/igbinary/commit/ddc68d59cdd784f538942f650c3e8d84a1db10ff. I'll ensure that's referenced on the PHPC ticket for context when we're able to start on it.

**Comment by mlocati** [comment]
Sorry about the mess, I was on a hurry...
